File: wallendar-app/app/src/main/java/com/td/wallendar/group/GroupHistoryAdapter.java
package com.td.wallendar.group;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.td.wallendar.R;
import com.td.wallendar.models.GroupHistory;

import java.util.ArrayList;
import java.util.List;

public class GroupHistoryAdapter extends RecyclerView.Adapter<GroupHistoryViewHolder> {

    private final List<GroupHistory> dataset;

    public GroupHistoryAdapter() {
        dataset = new ArrayList<>();
    }

    @NonNull
    @Override
    public GroupHistoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_group_activity, parent, false);
        return new GroupHistoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull GroupHistoryViewHolder holder, int position) {
        holder.bind(dataset.get(position));
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setDataset(final List<GroupHistory> newDataset) {
        this.dataset.clear();
        dataset.addAll(newDataset);
        notifyDataSetChanged();
    }

    public void addToDataset(final GroupHistory history) {
        this.dataset.add(0, history);
        notifyItemInserted(0);
    }

    @Override
    public int getItemCount() {
        return dataset == null ? 0 : dataset.size();
    }
}

File: wallendar-app/app/src/main/java/com/td/wallendar/group/GroupHistoryAdapter.kt
package com.td.wallendar.group

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.td.wallendar.R
import com.td.wallendar.models.GroupHistory
import java.util.*

class GroupHistoryAdapter : RecyclerView.Adapter<GroupHistoryViewHolder?>() {
    private val dataset: MutableList<GroupHistory>
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GroupHistoryViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_group_activity, parent, false)
        return GroupHistoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: GroupHistoryViewHolder, position: Int) {
        holder.bind(dataset[position])
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setDataset(newDataset: MutableList<GroupHistory>) {
        dataset.clear()
        dataset.addAll(newDataset)
        notifyDataSetChanged()
    }

    fun addToDataset(history: GroupHistory) {
        dataset.add(0, history)
        notifyItemInserted(0)
    }

    override fun getItemCount(): Int {
        return dataset.size
    }

    init {
        dataset = ArrayList()
    }
}